src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any, Union
+
+import fsspec
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, provider, credentials: dict[str, Any] = {}):
+        self.fs = FileStorageHelper.file_storage_init(
+            provider=provider, credentials=credentials
+        )
+
+    def read_file(
+        self,
+        path: str,
+        mode: str,
+        encoding: str = Common.DEFAULT_ENCODING,
+        seek_position: int = 0,
+        length: int = Common.FULL,
+    ) -> Union[bytes, str]:
+        """Read the file pointed to by the file_handle.
+
+        Args:
+            path (str): Path to the file to be opened
+            mode (str): Mode in whicg the file is to be opened. Usual options
+                        include r, rb, w and wb
+            encoding (str): Encoding type like utf-8 or utf-16
+            seek_position (int): Position to start reading from
+            length (int): Number of bytes to be read. Default is full
+            file content.
+
+        Returns:
+            Union[bytes, str] - File contents in bytes/string based on the opened mode
+        """
+        try:
+            with self.fs.open(path=path, mode=mode, encoding=encoding) as file_handle:
+                if seek_position > 0:
+                    file_handle.seek(seek_position)
+                return file_handle.read(length)
```

Comment:
@hari-kuriakose I think the read will load the complete file in to the memory. One way to avoid this is using read lines and yielding each line. But again at the downstream where ever it is being used if the data is not handled in proper manner entire read data could be loaded in to the memory. And in our case I don't know if implementing such a thing will help because I anticipate that all the downstream usage will be needing the entire data in memory.